analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -0,0 +1,284 @@
+Many of these have [very low ](#vaf-distributions) suggesting that some may be sequencing errors. 
+
+- Mutect2, Strelka2, and Lancet do have [plenty of mutations in common](#upset-graph). 
+
+### Outline of analyses completed:
+
+- [Upset graph](#upset-graph)  
+- [VAF of combinations of callers](#vaf-for-combination-sets-of-callers)
+- [Base changes by caller](#base-changes)
+- [VAF distributions by caller](#vaf-distributions)
+- [TMB distributions by caller](#tmb-distributions)
+- [VAF correlations between callers](#vaf-correlations)
+- [TMB correlations between callers](#tmb-correlations)
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/compare_snv_callers_plots.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r}
+if (!("ggupset" %in% installed.packages())) {
+  install.packages("ggupset", repos = "http://cran.us.r-project.org")
+}
+if (!("UpSetR" %in% installed.packages())) {
+  install.packages("UpSetR", repos = "http://cran.us.r-project.org")
+}
+if (!("GGally" %in% installed.packages())) {
+  install.packages("GGally", repos = "http://cran.us.r-project.org")
+}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Set up output directories. 
+
+```{r}
+base_results_dir <- "results"
+base_plots_dir <- "plots"
+```
+
+Make new directories for the comparison analysis.
+
+```{r}
+consensus_results_dir <- file.path(base_results_dir, "consensus")
+consensus_plots_dir <- file.path(base_plots_dir, "consensus")
+
+# Make caller specific plots folder
+if (!dir.exists(consensus_results_dir)) {
+  dir.create(consensus_results_dir)
+}
+# Make caller specific plots folder
+if (!dir.exists(consensus_plots_dir)) {
+  dir.create(consensus_plots_dir)
+}
+```
+
+## Import data 
+
+Load in the merged data.frames from `03-merge_callers.R` output. 
+
+```{r}
+# Master VAF data.frame
+vaf_df <- readr::read_rds(file.path(consensus_results_dir, 
+                                    "all_callers_vaf.rds"))
+
+# Master TMB data.frame
+tmb_df <- readr::read_rds(file.path(consensus_results_dir, 
+                                    "all_callers_tmb.rds"))
+
+# Full mutation list
+mutation_id_list <- readr::read_rds(file.path(consensus_results_dir, 
+                                              "mutation_id_list.rds"))
+
+# Callers per mutation data.frame
+callers_per_mutation <- readr::read_rds(file.path(consensus_results_dir, 
+                                                  "callers_per_mutation.rds"))
+```
+
+## Upset graph
+
+Make the upset plot. 
+
+```{r}
+# Make the plot
+upset <- UpSetR::upset(UpSetR::fromList(mutation_id_list),
+  order.by = "freq",
+  text.scale = 2
+)
+
+# Save this plot
+png(file.path(consensus_plots_dir, "upset_plot.png"), width = 900, height = 900)
+upset
+dev.off()
+
+# Print this out
+upset
+```
```

Comment:
This plot does not show up clearly on the html output. Perhaps try changing the figure dimensions here?

Reply:
Are you referring to the font size? Because the problem is a certain font size is too small for the png, but that same font size is too big for the notebook.